aspnet/fundamentals/error-handling.rst

```diff
@@ -1,8 +1,181 @@
+
+#. Once the headers for a response have been sent, you cannot change the response's status code.
+#. If the client disconnects mid-response, you cannot send them the rest of the content of that response.
+#. There is always the possibility of an error occuring one layer below your error handling layer.
+#. Don't forget, error handling pages can have errors, too. It's often a good idea for production error pages to consist of purely static content.
+
+Following the above recommendations will help ensure your app remains responsive and is able to gracefully handle errors that may occur.
+
+Server Error Handling
+---------------------
+
+In addition to ASP.NET, the server hosting your app will perform some error handling. In the case of IIS, this is described `here <https://technet.microsoft.com/en-us/library/cc731570(v=ws.10).aspx>`. For Kestrel, its built-in error handling behavior may be found `here <https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs#L565>`. Requests that are not handled by your app will be handled by the server, and any error that occurs will be handled by the server's error handling. Any custom error pages or error handling middleware or filters you have configured for your app will not affect this behavior.
+
+.. _startup-error-handling:
+
+Startup Error Handling
+----------------------
+
+One of the trickiest places to handle errors in your app is during its startup. Depending on how far into startup the app makes it before the error occurs, error handling middleware, filters, etc. may or may not be set up to deal with the problem. Errors that occur in your app's startup can also impact server behavior. For example, to enable SSL in Kestrel, one must configure the server with ``app.UseKestrelHttps()``. If an exception happens before this line in ``Startup``, then by default Kestrel will not handle the exception and the process will crash.
+
+If you wish to change this behavior (you would prefer the server start, and display the exception in response to requests), you can do so using ``CaptureStartupErrors``.
+
+.. code-block:: c#
+	:emphasize-lines: 2
+
+	var builder = new WebHostBuilder()
+		.CaptureStartupErrors(true)
+		.Configure(app => { ... });
+
+.. note:: This behavior was updated between RC1 and RC2.
+
+HTTP 500 Errors on Azure
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+If your app throws an exception before the ``Configure`` method in *Startup.cs* completes, the developer exception page won't be configured. The app deployed to Azure (or another production IIS server) will return an HTTP 500 error with no message details.
+
+The publish wizard in Visual Studio 2015 creates a *web.config* file if you don't have one. If you have a *web.config* file in the *wwwroot* folder, the deploy process inserts the markup into the *web.config* file it generates. 
```

Comment:
This should all be RC1 with the exception of the note on line 143 (behavior is updated in RC2); should I revert that for consistency's sake?